### PR TITLE
Added apply-to object reference to decals

### PR DESCRIPTION
**Pull request by jankrassnigg**
Also fixed how scaling is applied when the projection axis is not +X.
And fixed a crash in object picking.

### CODE REVIEW

File: Code/Engine/RendererCore/Decals/Implementation/DecalComponent.cpp [hunk 1]
```diff
@@ -401,6 +401,54 @@ void ezDecalComponent::OnMsgExtractRenderData(ezMsgExtractRenderData& msg) const
...
+      {
+        m_uiApplyOnlyToId = pRenderComponent->GetUniqueIdForRendering();
+      }
+    }
+  }
+
+  if (uiPrevId != m_uiApplyOnlyToId && GetOwner()->IsStatic())
+  {
+    // TODO: clear render data cache ?
+    TriggerLocalBoundsUpdate();
```
Comment (C-Core) [inline Code/Engine/RendererCore/Decals/Implementation/DecalComponent.cpp]: Why do we trigger a bounds update here?

**Comment by jankrassnigg** [comment]
Ah right, I want to make sure for static objects the render data cache is
invalidated, could not find a better way.

C-Core <notifications@github.com> schrieb am Di., 31. März 2020, 21:39:

File: Code/Engine/RendererCore/Decals/Implementation/DecalComponent.cpp (hunk 1 above)
Comment (jankrassnigg) [reply Code/Engine/RendererCore/Decals/Implementation/DecalComponent.cpp]: fixed

File: Code/Engine/RendererCore/Meshes/Implementation/MeshComponentBase.cpp [hunk 2]
```diff
@@ -192,13 +192,19 @@ void ezMeshComponentBase::OnMsgExtractRenderData(ezMsgExtractRenderData& msg) co
...
+
     // Determine render data category.
     ezRenderData::Category category = m_RenderDataCategory;
     if (category == ezInvalidRenderDataCategory)
     {
       if (hMaterial.IsValid())
       {
         ezResourceLock<ezMaterialResource> pMaterial(hMaterial, ezResourceAcquireMode::AllowLoadingFallback);
+
+        if (pMaterial.GetAcquireResult() == ezResourceAcquireResult::LoadingFallback)
```
Comment (jankrassnigg) [inline Code/Engine/RendererCore/Meshes/Implementation/MeshComponentBase.cpp:205]: @C-Core kind of an important fix. I already fixed this in the greybox component previously, but this way is more elegant IMO, so I applied the same fix there now.

**Pull request merged**